lib/index.js

```diff
@@ -290,6 +300,32 @@ Server.prototype.serve = function(req, res){
 };
 
 /**
+ * Handles a request serving `/socket.io.js.map`
+ *
+ * @param {http.Request} req
+ * @param {http.Response} res
+ * @api private
+ */
+
+Server.prototype.serveMap = function(req, res){
+  var etag = req.headers['if-none-match'];
+  if (etag) {
+    if (clientVersion == etag) {
+      debug('serve client 304');
+      res.writeHead(304);
+      res.end();
+      return;
+    }
+  }
+
+  debug('serve client sourcemap');
+  res.setHeader('Content-Type', 'application/javascript');
```

Comment:
Also, according to [this](http://stackoverflow.com/questions/19911929/what-mime-type-should-i-use-for-javascript-source-map-files), it seems the content-type should rather be `application/json`.